packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -162,6 +163,13 @@ class BottomSheet extends StatefulWidget {
   /// will be [Clip.none].
   final Clip? clipBehavior;
 
+  /// Defines minimum and maximum sizes for a [BottomSheet].
+  ///
+  /// If null, then the ambient [ThemeData.bottomSheetTheme]'s
+  /// [BottomSheetThemeData.constraints] will be used. If that
+  /// is null then the bottom sheet's size will be unconstrained.
+  final BoxConstraints? constraints;
```

Comment:
There are some subtle things going on here in typical use cases. Typically the incoming constraint for the bottom-sheet's maximum width will be as wide as the enclosing route/window/screen. If this constraints object specifies a smaller maximum width, then the bottom-sheet will be aligned within the wider horizontal space specified by the incoming constraint.  That's going to be the typical scenario on devices with big screens. I think it would be worth explaining this a bit.